### PR TITLE
[Feature][seatunnel-examples] flink  local environment run quickly and debug locally developed code easily

### LINKED ISSUE #955: [Feature][seatunnel-test] flink local test
### Search before asking

- [X] I had searched in the [feature](https://github.com/apache/incubator-seatunnel/issues?q=is%3Aissue+label%3A%22Feature%22) and found no similar feature requirement.

### Description

Debugging and development is not easy at the moment。
Provide a local debug environment for development。

### Usage Scenario

Provide a local debug environment for development

### Related issues

_No response_

### Are you willing to submit a PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by felix-thinkingdata**
this closes #955

## Check list

* [x] Code changed are covered with tests, or it does not need tests for reason:
* [x] If necessary, please update the documentation to describe the new feature. https://github.com/apache/incubator-seatunnel/tree/dev/docs

### CODE REVIEW

**Comment by leo65535** [comment]
Seems we can rename `seatunnel-test` to `seatunnel-examples`

**Comment by felix-thinkingdata** [comment]
image


> Seems we can rename `seatunnel-test` to `seatunnel-examples`

ok

**Comment by leo65535** [comment]
+1 this is a good idea to instroduce a new module `seatunnel-examples` to help users to run the demo quickly.

**Comment by leo65535** [comment]
`SeaTunnel` supports `flink` and `spark` engine, it's better add two sub module under `seatunnel-examples`.

**Comment by felix-thinkingdata** [comment]
> `SeaTunnel` supports `flink` and `spark` engine, it's better add two sub module under `seatunnel-examples`.



> `SeaTunnel` supports `flink` and `spark` engine, it's better add two sub module under `seatunnel-examples`.

Spark is not very familiar, SO I will look into it

**Comment by felix-thinkingdata** [comment]
> > `SeaTunnel` supports `flink` and `spark` engine, it's better add two sub module under `seatunnel-examples`.
> 
> > `SeaTunnel` supports `flink` and `spark` engine, it's better add two sub module under `seatunnel-examples`.
> 
> Spark is not very familiar, SO I will look into it

image
image

File: seatunnel-examples/seatunnel-flink-examples/pom.xml [hunk 1]
```diff
@@ -0,0 +1,67 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project xmlns="http://maven.apache.org/POM/4.0.0"
```
Comment (wuchunfu) [inline seatunnel-examples/seatunnel-flink-examples/pom.xml:2]: Please add license, thanks

Comment (felix-thinkingdata) [reply seatunnel-examples/seatunnel-flink-examples/pom.xml:2]: > Please add license, thanks

ok

File: seatunnel-examples/pom.xml [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+    <modelVersion>4.0.0</modelVersion>
+
+    <artifactId>seatunnel-examples</artifactId>
+    <packaging>pom</packaging>
+    <modules>
+        <module>seatunnel-flink-examples</module>
+    </modules>
+
+
+
```
Comment (wuchunfu) [inline seatunnel-examples/pom.xml:37]: Please check it as a whole and remove useless spaces, thanks

File: seatunnel-examples/seatunnel-flink-examples/pom.xml [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+            <groupId>org.apache.flink</groupId>
+            <artifactId>flink-streaming-java_${scala.binary.version}</artifactId>
+            <version>${flink.version}</version>
+            <scope>${flink.scope}</scope>
+        </dependency>
+        <dependency>
+            <groupId>org.slf4j</groupId>
+            <artifactId>slf4j-simple</artifactId>
+            <version>1.7.25</version>
+        </dependency>
```
Comment (wuchunfu) [inline seatunnel-examples/seatunnel-flink-examples/pom.xml:81]: What is the purpose of adding this dependency separately?

File: seatunnel-examples/seatunnel-flink-examples/src/main/resources/examples/fake_to_console.config [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+  # please go to https://interestinglab.github.io/seatunnel-docs/#/zh-cn/configuration/base
+}
+
+sink {
+  ConsoleSink {}
+
+
+  # If you would like to get more information about how to configure seatunnel and see full list of sink plugins,
+  # please go to https://interestinglab.github.io/seatunnel-docs/#/zh-cn/configuration/base
+}
```
Comment (wuchunfu) [inline seatunnel-examples/seatunnel-flink-examples/src/main/resources/examples/fake_to_console.config:38]: Do you want to enable the new official document address for the document address here?

File: seatunnel-examples/seatunnel-flink-examples/pom.xml (hunk 3 above)
Comment (felix-thinkingdata) [reply seatunnel-examples/seatunnel-flink-examples/pom.xml:81]: If not added, the following exceptions will occur:
`SLF4J: Failed to load class "org.slf4j.impl.StaticLoggerBinder".
SLF4J: Defaulting to no-operation (NOP) logger implementation`



I'll see if there's a better way

File: seatunnel-examples/seatunnel-flink-examples/src/main/resources/examples/fake_to_console.config (hunk 4 above)
Comment (felix-thinkingdata) [reply seatunnel-examples/seatunnel-flink-examples/src/main/resources/examples/fake_to_console.config:38]: Let me change it. I copied it from the demo. I forgot to change it

File: seatunnel-examples/pom.xml (hunk 2 above)
Comment (felix-thinkingdata) [reply seatunnel-examples/pom.xml]: > Please check it as a whole and remove useless spaces, thanks

fix

File: seatunnel-examples/seatunnel-flink-examples/pom.xml (hunk 3 above)
Comment (felix-thinkingdata) [reply seatunnel-examples/seatunnel-flink-examples/pom.xml]: Resolved ,the addition of log4j.properties

**Comment by felix-thinkingdata** [comment]
@wuchunfu It's been fixed ，Can you review it for me? Thank you

**Comment by yx91490** [comment]
Why not use docker to implement local test https://github.com/apache/incubator-seatunnel/pull/815 ?

**Comment by CalvinKirs** [comment]
> Why not use docker to implement local test #815 ?

We always hope that users can quickly test in various environments. For developers, this is a good experience.

**Comment by yx91490** [comment]
> > Why not use docker to implement local test #815 ?
> 
> We always hope that users can quickly test in various environments. For developers, this is a good experience.

It's too expensive to create a new module, maybe add as a unit test instead?

**Comment by RickyHuo** [comment]
`seatunnel-flink-examples/src/main/resources/examples/fake_to_console.config`  can replaced with `config/flink.streaming.conf.template`

**Comment by CalvinKirs** [comment]
> > > Why not use docker to implement local test #815 ?
> > 
> > 
> > We always hope that users can quickly test in various environments. For developers, this is a good experience.
> 
> It's too expensive to create a new module, maybe add as a unit test instead?



> > > Why not use docker to implement local test #815 ?
> > 
> > 
> > We always hope that users can quickly test in various environments. For developers, this is a good experience.
> 
> It's too expensive to create a new module, maybe add as a unit test instead?

Sounds good~ WDYT ？@felix-thinkingdata

**Comment by felix-thinkingdata** [comment]
> flink.streaming.conf.template

You need to change Maven Scope to Runtime or compile to run properly.If put in unit test, need to change the scope of the entire project Maven, not very suitable

**Comment by felix-thinkingdata** [comment]
> > > > Why not use docker to implement local test #815 ?
> > > 
> > > 
> > > We always hope that users can quickly test in various environments. For developers, this is a good experience.
> > 
> > 
> > It's too expensive to create a new module, maybe add as a unit test instead?
> 
> > > > Why not use docker to implement local test #815 ?
> > > 
> > > 
> > > We always hope that users can quickly test in various environments. For developers, this is a good experience.
> > 
> > 
> > It's too expensive to create a new module, maybe add as a unit test instead?
> 
> Sounds good~ WDYT ？@felix-thinkingdata
You need to change Maven Scope to Runtime or compile to run properly.If put in unit test, need to change the scope of the entire project Maven, not very suitable

**Comment by leo65535** [comment]
from my side, an independent may better, like other projects
https://github.com/apache/spark/tree/master/examples
https://github.com/apache/flink/tree/master/flink-examples
https://github.com/apache/hudi/tree/master/hudi-examples

**Comment by felix-thinkingdata** [comment]
> from my side, an independent may better, like other projects https://github.com/apache/spark/tree/master/examples https://github.com/apache/flink/tree/master/flink-examples https://github.com/apache/hudi/tree/master/hudi-examples

Is there anything else I need to change？

File: seatunnel-examples/seatunnel-flink-examples/src/main/resources/examples/fake_to_console.config [hunk 5]
```diff
@@ -0,0 +1,55 @@
+##
+## Licensed to the Apache Software Foundation (ASF) under one or more
+## contributor license agreements.  See the NOTICE file distributed with
+## this work for additional information regarding copyright ownership.
+## The ASF licenses this file to You under the Apache License, Version 2.0
+## (the "License"); you may not use this file except in compliance with
+## the License.  You may obtain a copy of the License at
+##
+##    http://www.apache.org/licenses/LICENSE-2.0
```
Comment (CalvinKirs) [inline seatunnel-examples/seatunnel-flink-examples/src/main/resources/examples/fake_to_console.config:9]: Please keep the license format consistent

Comment (felix-thinkingdata) [reply seatunnel-examples/seatunnel-flink-examples/src/main/resources/examples/fake_to_console.config]: > Please keep the license format consistent

done

**Review by CalvinKirs — APPROVED** [review]
LGTM

**Comment by CalvinKirs** [comment]
@leo65535 PTAL,thx ~

**Comment by yx91490** [comment]
> from my side, an independent may better, like other projects https://github.com/apache/spark/tree/master/examples https://github.com/apache/flink/tree/master/flink-examples https://github.com/apache/hudi/tree/master/hudi-examples

It seems that these modules don't aim at local test.

**Comment by felix-thinkingdata** [comment]
> 

The main solution is to allow developers to run quickly and debug locally developed code easily.

**Review by leo65535 — APPROVED** [review]
LGTM, please update the title of the patch

**Comment by felix-thinkingdata** [comment]
> LGTM, please update the title of the patch

done ,Is the current title ok？

**Comment by leo65535** [comment]
> done ,Is the current title ok？

💯

**Pull request merged**